lib/erb_lint/linters/hard_coded_string.rb

```diff
@@ -1,32 +1,74 @@
 # frozen_string_literal: true
 
 require 'better_html/tree/tag'
+require 'active_support/core_ext/string/inflections'
 
 module ERBLint
   module Linters
     # Checks for hardcoded strings. Useful if you want to ensure a string can be translated using i18n.
     class HardCodedString < Linter
       include LinterRegistry
 
+      MissingCorrector = Class.new(StandardError)
+
+      class ConfigSchema < LinterConfig
+        property :corrector, accepts: Hash, required: false, default: {}
+      end
+      self.config_schema = ConfigSchema
+
       def offenses(processed_source)
         hardcoded_strings = processed_source.ast.descendants(:text).each_with_object([]) do |text_node, to_check|
           next if javascript?(processed_source, text_node)
 
-          offended_str = text_node.to_a.find { |node| relevant_node(node) }
-          to_check << [text_node, offended_str] if offended_str
+          offended_strings = text_node.to_a.select { |node| relevant_node(node) }
+          offended_strings.each do |offended_string|
+            offended_string.split("\n").each do |str|
+              to_check << [text_node, str] unless str.empty?
+            end
+          end
         end
 
         hardcoded_strings.compact.map do |text_node, offended_str|
+          range_begin, range_stop = find_range(text_node, offended_str)
+          source_range = processed_source.to_source_range(range_begin, range_stop)
+
           Offense.new(
             self,
-            processed_source.to_source_range(text_node.loc.start, text_node.loc.stop),
-            message(offended_str)
+            source_range,
+            message(source_range.source)
           )
         end
       end
 
+      def find_range(node, str)
+        match = node.loc.source.match(Regexp.new(Regexp.quote(str.strip)))
+        return unless match
+
+        range_begin = match.begin(0) + node.loc.start
+        range_end   = match.end(0) + node.loc.start - 1
+        [range_begin, range_end]
+      end
+
+      def autocorrect(processed_source, offense)
+        string = offense.source_range.source
+        return unless klass = load_corrector
+        return unless string.strip.length > 1
+
+        corrector = klass.new(processed_source.filename, offense.source_range)
+        node = RuboCop::AST::StrNode.new(:str, [string])
+        corrector.autocorrect(node, tag_start: '<%= ', tag_end: ' %>')
+      rescue MissingCorrector
+        nil
+      end
+
       private
 
+      def load_corrector
+        require @config['corrector'].fetch('path') { raise MissingCorrector }
+
+        @config['corrector'].fetch('name').safe_constantize
```

Comment:
I don't think we can do this. policial.shopify.io makes public repositories viewable to any github user, and runs on public repositories as well. I believe anyone would be able to create a PR and change the config file in their branch, policial would pick that up and instanciate arbitrary objects. This is one step too close to remote code execution. Can you make this a whitelist, or an explicit dependency?

Reply:
but policial.io doesn't run erb-lint with the autocorrection flag, does it ?

Reply:
Not yet but I am working on that feature :/